src/test/clojure/clara/test_rules.clj

```diff
@@ -2748,3 +2748,19 @@
                     (fire-rules))]
 
     (is (= [{:?x inner-one}] (query session test-query)))))
+
+;; Ensure that nil values can be bound and unified as expected.
```

Comment:
Perhaps this comment should refer to [issue 393](https://github.com/cerner/clara-rules/issues/393) as well since it seems fairly specific to it as it has the nested record pattern.